pkg/appmanager/resourceConfig_test.go

```diff
@@ -937,4 +937,436 @@ var _ = Describe("Resource Config Tests", func() {
+
+			result = parseAppRootURLRewriteAnnotations(",")
+			Expect(len(result)).To(Equal(0))
+
+			result = parseAppRootURLRewriteAnnotations("foo,")
+			Expect(len(result)).To(Equal(0))
+
+			result = parseAppRootURLRewriteAnnotations("foo=bar,")
+			Expect(len(result)).To(Equal(1))
+			Expect(result["foo"]).To(Equal("bar"))
+
+			result = parseAppRootURLRewriteAnnotations("foo=bar,bar=baz,")
+			Expect(len(result)).To(Equal(2))
+			Expect(result["foo"]).To(Equal("bar"))
+			Expect(result["bar"]).To(Equal("baz"))
+
+			result = parseAppRootURLRewriteAnnotations("foo=bar,bar")
+			Expect(len(result)).To(Equal(1))
+			Expect(result["foo"]).To(Equal("bar"))
+
+			result = parseAppRootURLRewriteAnnotations("foo,bar=baz")
+			Expect(len(result)).To(Equal(1))
+			Expect(result["bar"]).To(Equal("baz"))
+
+			result = parseAppRootURLRewriteAnnotations("foo=bar,=bar=")
+			Expect(len(result)).To(Equal(1))
+			Expect(result["foo"]).To(Equal("bar"))
+
+			result = parseAppRootURLRewriteAnnotations(",bar=baz")
+			Expect(len(result)).To(Equal(1))
+			Expect(result["bar"]).To(Equal("baz"))
+
+			result = parseAppRootURLRewriteAnnotations("/hello=/good-bye/hi=/bye")
+			Expect(len(result)).To(Equal(0))
+
+			result = parseAppRootURLRewriteAnnotations("/hello=/good-bye/hi=/bye,")
+			Expect(len(result)).To(Equal(0))
+
+			result = parseAppRootURLRewriteAnnotations(",/hello=/good-bye/hi=/bye,")
+			Expect(len(result)).To(Equal(0))
+		})
+
+		It("processes url-rewrite annotations correctly", func() {
+			// Non-targeted url-rewrite rewrite host
+			rule := processURLRewrite("", "host.com")
+			Expect(len(rule.Conditions)).To(Equal(0))
+			Expect(len(rule.Actions)).To(Equal(1))
+			Expect(rule.Actions[0].Name).To(Equal("0"))
+			Expect(rule.Actions[0].HTTPHost).To(Equal(true))
+			Expect(rule.Actions[0].Replace).To(Equal(true))
+			Expect(rule.Actions[0].Value).To(Equal("host.com"))
+
+			// Non-targeted url-rewrite rewrite path
+			rule = processURLRewrite("", "/path")
+			Expect(len(rule.Conditions)).To(Equal(0))
+			Expect(len(rule.Actions)).To(Equal(1))
+			Expect(rule.Actions[0].Name).To(Equal("0"))
+			Expect(rule.Actions[0].HTTPURI).To(Equal(true))
+			Expect(rule.Actions[0].Path).To(Equal(""))
+			Expect(rule.Actions[0].Replace).To(Equal(true))
+			Expect(rule.Actions[0].Value).To(Equal("/path"))
+
+			// Non-targeted url-rewrite rewrite host and path
+			rule = processURLRewrite("", "host.com/path")
+			Expect(len(rule.Conditions)).To(Equal(0))
+			Expect(len(rule.Actions)).To(Equal(2))
+			Expect(rule.Actions[0].Name).To(Equal("0"))
+			Expect(rule.Actions[0].HTTPHost).To(Equal(true))
+			Expect(rule.Actions[0].Replace).To(Equal(true))
+			Expect(rule.Actions[0].Value).To(Equal("host.com"))
+			Expect(rule.Actions[1].Name).To(Equal("1"))
+			Expect(rule.Actions[1].HTTPURI).To(Equal(true))
+			Expect(rule.Actions[1].Path).To(Equal(""))
+			Expect(rule.Actions[1].Replace).To(Equal(true))
+			Expect(rule.Actions[1].Value).To(Equal("/path"))
```

Comment:
Seems a bit overkill to be testing every value in the actions for all of these test cases. I would say the only values we care about are Name and Value (for both actions and conditions). If those values equal what we expect, then we know all other values are "true", since they can't be false unless the entire rule is uninitialized.

Reply:
I like being this explicit in this test since values are used for both HTTPURI and HTTPHost so if this function was written incorrectly you could get a rule of HTTPHost: true, Values: [/path] which is incorrect but the test would still pass since we only check name and values.